cmd/jaeger/internal/integration/kafka_test.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package integration
+
+import (
+	"testing"
+
+	"github.com/jaegertracing/jaeger/plugin/storage/integration"
+)
+
+func TestKafkaStorage(t *testing.T) {
+	integration.SkipUnlessEnv(t, "kafka")
+
+	collectorConfig := "../../collector-with-kafka.yaml"
+	ingesterConfig := "../../ingester-remote-storage.yaml"
+
+	collector := &E2EStorageIntegration{
+		SkipStorageCleaner: true,
+		ConfigFile:         collectorConfig,
+	}
+
+	// Initialize and start the collector
+	collector.e2eInitialize(t, "kafka")
+
+	ingester := &E2EStorageIntegration{
+		ConfigFile: ingesterConfig,
+		StorageIntegration: integration.StorageIntegration{
+			CleanUp:                      purge,
+			GetDependenciesReturnsSource: true,
+			SkipArchiveTest:              true,
+		},
+	}
+
+	// Initialize and start the ingester
+	ingester.e2eInitialize(t, "kafka")
+
+	// Set up cleanup for both collector and ingester
+	t.Cleanup(func() {
+		collector.e2eCleanUp(t)
```

Comment:
Small nit: cleanup needs to be scheduled as soon as the component is started. Otherwise you may start collector, fail to start ingester, and never reach this point to schedule cleanup of the collector. 

An even better solution would be for cleanup to be scheduled by initialize function in the first place, so that we don't have to do it manually in every test.

Reply:
Okay understood, on it.